wordpress-mobile/WordPress-Android
# [Reader Improvements] Update Site Header to match i2 design

Fixes #19069 

To test:
1 - Open `"Me"` -> Debug settings -> `"Features in development"`;
2 - Enable `ReaderImprovementsFeatureConfig`;
3 - Move back to home and open Reader;
4 - Tap on any site's name/domain to open site posts list;
5 - You should see the new header layout. Make sure follow/unfollow button is working as expected;
<p float="left">
<img width="300" src="https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/65e553f9-33be-4f0b-a1d8-a9d8439b4722">
<img width="300" src="https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/8ca4ec24-90b6-4efb-8426-ab9a93ab66ab">
</p>

6 - Select a site without icon: the icon should be gone instead of displaying the placeholder;
7 - Repeat step 1 but this time disable `ReaderImprovementsFeatureConfig`;
8 - Repeat steps 3 and 4: you should see the existing header layout. Make sure follow/unfollow button is working as expected.
<p float="left">
<img width="300" alt="Screen Shot 2023-08-25 at 10 57 09 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/ab1e9d89-24db-437a-9e98-306adcecad3b">
<img width="300" alt="Screen Shot 2023-08-25 at 11 11 12 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/6acca4be-3477-49fa-aea8-5db7419f9c99">
</p>


## Regression Notes
1. Potential unintended areas of impact
None (implementation behind feature flag)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None (all existing logic is inside a custom view)

PR submission checklist:


UI Changes testing checklist:
